src/track/serato/beatgrid.cpp

```diff
@@ -440,21 +440,23 @@ void SeratoBeatGrid::setBeats(BeatsPointer pBeats,
         return;
     }
 
-    SINT currentBeatPositionSamples = 0;
```

Comment:
is std::ceil(streamInfo.getDuration().toDoubleSeconds())))); above correct? I think I would remove std::ceil

Reply:
I wanted to make sure that I capture *all* beats in the track, so I rounded up. The API is a bit clunky when I try to get an iterator over all available beats :(

Reply:
Isn't the current comment sufficiently clear?

```cpp
    // Find at least the track duration in samples. This value might be longer
    // than the actual track, to make sure we get all beats.
    const SINT trackDurationSamples = streamInfo.getSignalInfo().frames2samples(
            static_cast<SINT>(streamInfo.getSignalInfo().secs2frames(
                    std::ceil(streamInfo.getDuration().toDoubleSeconds()))));
```

Reply:
After your additional explanations yes. When I original read it I did not connect it to the std::ceil() call. Can you reference it explicitly and that this is a workaround for an internal rounding issue or such?